### PR TITLE
Code optimizations reported by static code analysis (2018-11-08)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: Performance "Code optimizations reported by static code analysis"

#### Purpose of change

Cleanup code while increase performance and fix minor error.

#### Describe the solution

See commit messages.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -1323,7 +1323,7 @@ bool game::cleanup_at_end()
                     case 2:
                         queryDelete = true;
                         break;
-                };
+                }s
```
Comment (AMurkin) [inline src/game.cpp]: `s`?

Comment (ZhilkinSerg) [reply src/game.cpp]: Typo. Will push fix soon.

File: src/animation.cpp [hunk 2]
```diff
@@ -102,7 +102,7 @@ tripoint relative_view_pos( player const &u, tripoint const &p ) noexcept
 }
 
 // Convert p to screen position relative to the current terrain view
-static tripoint relative_view_pos( game const &g, tripoint const &p ) noexcept
+tripoint relative_view_pos( game const &g, tripoint const &p ) noexcept
```
Comment (BevapDin) [inline src/animation.cpp]: What is the reason for that change?

File: src/cata_tiles.cpp [hunk 3]
```diff
@@ -952,11 +952,9 @@ void tileset_loader::load_tilejson_from_file( JsonObject &config )
  * The JSON data (loaded here) contains tile ids relative to the associated image.
  * They are translated into global ids by adding the @p offset, which is the number of
  * previously loaded tiles (excluding the tiles from the associated image).
+ * @param entry
```
Comment (BevapDin) [inline src/cata_tiles.cpp]: And why that? If you don't add any useful information, don't add a line for it.

File: src/regional_settings.h [hunk 4]
```diff
@@ -22,7 +22,7 @@ class building_bin
         weighted_int_list<overmap_special_id> buildings;
         std::map<overmap_special_id, int> unfinalized_buildings;
     public:
-        building_bin() {};
+        building_bin() {}
```
Comment (BevapDin) [inline src/regional_settings.h]: You can default those constructors: `building_bin() = default;` - similar in other places.

Comment (ZhilkinSerg) [reply src/regional_settings.h]: Thanks. I've did that were applicable.

Comment (ZhilkinSerg) [reply src/regional_settings.h]: Hmm.. It seems clang doesn't like this.

Comment (ZhilkinSerg) [reply src/regional_settings.h]: @narc0tiq: What's clang version on Jenkins PR builder? Is it 3.8?

Comment (narc0tiq) [reply src/regional_settings.h]: Should've been in the build output, but I think I ballsed that up at some revision of the mainline build script. But yes:
```
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

Comment (jbytheway) [reply src/regional_settings.h]: The clang error relates specifically to the way in which `selection_column_preset` is used; the others should probably be fine.  (It also looks like a clang-3.8 bug).

Comment (ZhilkinSerg) [reply src/regional_settings.h]: We'll see now.

Comment (ZhilkinSerg) [reply src/regional_settings.h]: `overmap_terrain_special` was also used in the same way, so I had to update its constructor too.

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

**Pull request merged**